pyquil/quil.py

```diff
@@ -643,6 +643,14 @@ def address_qubits(program, qubit_mapping=None):
             result.append(Gate(instr.name, instr.params, remapped_qubits))
         elif isinstance(instr, Measurement):
             result.append(Measurement(qubit_mapping[instr.qubit], instr.classical_reg))
+        elif isinstance(instr, Pragma):
```

Comment:
I think it would be better to include Pragmas in the check for "what type of qubits does it use" infrastructure and then remap here in the same way that Gates and Measurements are done

Reply:
ya

Reply:
@mpharrigan @stevenheidel I think this might be a point of confusion for me. Pragmas can have numbers as arguments that represent things besides qubits. 

I assume that currently when I use a number to represent a qubit in a program it is converted internally into a qubit object. I think it's still meaningful to extend the current "what type of qubits does it use architecture", but it's going to be a bit different, as far as I can tell.